### PR TITLE
improve React exampless using `React.createRef()` to make them copy-pastable

### PR DESCRIPTION
**Pull request by maciejbo-handsontable**
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1.
2.
3.

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [x] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [] My change requires a change to the documentation.

[skip changelog]

### CODE REVIEW

File: docs/content/guides/columns/column-filter.md [hunk 1]
```diff
@@ -266,7 +266,7 @@ const hot = new Handsontable(container, {
 ::: only-for react
 ::: example #example3 :react
 ```jsx
-import React, { useEffect } from 'react';
+import React, { useEffect, useRef } from 'react';
```
Comment (wszymanski) [inline docs/content/guides/columns/column-filter.md:269]: It should work probably.

```suggestion
import { useEffect, useRef } from 'react';
```

File: docs/content/guides/columns/react-hot-column.md [hunk 2]
```diff
@@ -144,7 +144,7 @@ In this example, the custom editor component is created with an external depende
 
 ::: example #example6 :react-advanced --tab preview
 ```jsx
-import { useEffect } from 'react';
+import React, { useEffect } from 'react';
```
Comment (wszymanski) [inline docs/content/guides/columns/react-hot-column.md:147]: It should work probably.

```suggestion
import { useEffect, useRef } from 'react';
```

File: docs/content/guides/columns/react-hot-column.md [hunk 3]
```diff
@@ -158,7 +158,7 @@ class UnconnectedColorPicker extends BaseEditorComponent {
   constructor(props) {
     super(props);
 
-    this.editorRef = React.createRef();
+    this.editorRef = React.createRef(null);
```
Comment (wszymanski) [inline docs/content/guides/columns/react-hot-column.md:161]: Could it be changed with `useRef(null)`?

**Review by wszymanski — CHANGES_REQUESTED** [review]
I have some suggestions. Could you change also: `/docs/react-data-grid/cell-editor/#component-based-editors` page?

File: docs/content/guides/columns/react-hot-column.md (hunk 3 above)
Comment (maciejbo-handsontable) [reply docs/content/guides/columns/react-hot-column.md:161]: hey, @wszymanski - I checked it now once again, but unfortunatelly `useRef(null)` works correctly only in functional components and in this case your proposition won't work.

useref

File: docs/content/guides/columns/column-filter.md (hunk 1 above)
Comment (maciejbo-handsontable) [reply docs/content/guides/columns/column-filter.md]: that's true, looks like we do not need this import here.

**Review by wszymanski — CHANGES_REQUESTED** [review]
This suggestion still up to date: https://github.com/handsontable/handsontable/pull/9938#discussion_r978455307

File: docs/content/guides/columns/react-hot-column.md (hunk 2 above)
Comment (maciejbo-handsontable) [reply docs/content/guides/columns/react-hot-column.md:147]: hey, we need this `React` import here, cause in few next lines we use `React.createRef()` which can not be replaced with just `useRef(null)` in this example currently

Comment (wszymanski) [reply docs/content/guides/columns/react-hot-column.md:147]: Sure, thanks for explanation, I've missed that.

**Review by wszymanski — APPROVED** [review]

**Pull request merged**